sdk/physical/inmem/inmem.go

```diff
@@ -378,7 +378,18 @@ func (t *TransactionalInmemBackend) Transaction(ctx context.Context, txns []*phy
 		t.largestBatchLen.Store(uint64(len(txns)))
 	}
 
-	return physical.GenericTransactionHandler(ctx, t, txns)
+	err := physical.GenericTransactionHandler(ctx, t, txns)
+
+	// If we have a transactionCompleteCh set, we block on it before returning.
+	if t.transactionCompleteCh != nil {
```

Comment:
Is this change supposed to be in this PR?

Reply:
Yep, it's required for the tests in the Enterprise PR to make sure we can test the new WAL behaviour deterministically.